seed/data_importer/tasks.py

```diff
@@ -1119,30 +1120,101 @@ def save_raw_data(file_pk):
     return progress_data.result()
 
 
+def geocode_and_match_buildings_task(file_pk):
+    import_file = ImportFile.objects.get(pk=file_pk)
+
+    progress_data = ProgressData(func_name='match_buildings', unique_id=file_pk)
+    progress_data.delete()
+
+    if import_file.matching_done:
+        _log.debug('Matching is already done')
+        return progress_data.finish_with_warning('matching already complete')
+
+    if not import_file.mapping_done:
+        _log.debug('Mapping is not done yet')
+        return progress_data.finish_with_error(
+            'Import file is not complete. Retry after mapping is complete', )
+
+    post_geocode_tasks = None
+    if import_file.from_buildingsync:
+        if import_file.cycle is None:
+            _log.warn("This should never happen in production")
```

Comment:
Nbd, but from what I can tell in `match_buildings()` and `map_additional_models()` I think we want this check to happen for either case - BuildingSync and else.

Reply:
Nice catch